code/game/machinery/vr_console.dm

```diff
@@ -0,0 +1,222 @@
+/obj/machinery/vr_sleeper
+	name = "VR sleeper"
+	desc = "A fancy bed with built-in sensory I/O ports and connectors to interface users' minds with their bodies in virtual reality."
+	icon = 'icons/obj/Cryogenic2.dmi'
+	icon_state = "syndipod_0"
+	density = 1
+	anchored = 1
+	circuit = /obj/item/weapon/circuitboard/vr_sleeper
+	var/mob/living/carbon/human/occupant = null
+	var/mob/living/carbon/human/avatar = null
+	var/datum/mind/vr_mind = null
+
+	use_power = 1
+	idle_power_usage = 15
+	active_power_usage = 200
+	light_color = "#FF0000"
+
+/obj/machinery/vr_sleeper/New()
+	..()
+	component_parts = list()
+	component_parts += new /obj/item/weapon/stock_parts/scanning_module(src)
+	component_parts += new /obj/item/stack/material/glass/reinforced(src, 2)
+
+	RefreshParts()
+
+/obj/machinery/vr_sleeper/initialize()
+	update_icon()
+
+/obj/machinery/vr_sleeper/process()
+	if(stat & (NOPOWER|BROKEN))
+		return
+
+/obj/machinery/vr_sleeper/update_icon()
+	icon_state = "syndipod_[occupant ? "1" : "0"]"
+

```

Comment:
Should there not be a ..() in here somewhere?

Reply:
All of these were copied more or less wholesale from ABS and sleeper code, lack of super's included

Reply:
Don't think you need a super unless you need something in machinery to do something, which considering generally update_icon() does is change the icon and sometimes does lighting stuff, you probably don't need to use a super.